scripts/build_a_package.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh
+
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script builds wheels for a single package when trigged from a push to
+# a tag as part of a GitHub workflow (See .github/publish-a-package.yml). The
+# wheel is then published to PyPI.
+
+set -ev
+
+if [[ -z $GITHUB_REF ]]; then
+  echo 'Failed to run script, missing workflow env variable GITHUB_REF'
+  exit -1
+fi
+
+# Follows PEP 440 versioning conventions: https://www.python.org/dev/peps/pep-0440/#pre-releases
+# Only allow publishing packages that are 1.0+ using this method
+if ! [[ $GITHUB_REF =~ ^refs/tags/(opentelemetry-.*)==([1-9]([0-9]+)?\.[0-9]+((a|b|rc)[0-9]+)?(\.(post)?[0-9]+)?)$ ]]; then
+  echo 'Failed to parse package name and package version from tag.'
+  exit -1
+fi
+
+# From the capture groups in the regex above
+PKG_NAME=${BASH_REMATCH[1]}
+PKG_VERSION=${BASH_REMATCH[2]}
+
+# Get the latest versions of packaging tools
+python3 -m pip install --upgrade pip setuptools wheel
+
+# If testing locally, change `BASEDIR` to your own repo directory
+BASEDIR=$(dirname $(readlink -f $(dirname $0)))
+cd $BASEDIR
+
+DISTDIR=${BASEDIR}/dist
+mkdir -p $DISTDIR
+rm -rf $DISTDIR/*
+
+for DIRECTORY_WITH_PACKAGE in exporter/*/ instrumentation/*/ propagator/*/ sdk-extension/*/ util/*/; do
```

Comment:
Would something like this work?

```bash
SETUP_PY_FILE=*/*/$PKG_NAME/setup.py
DIRECTORY_WITH_PACKAGE=$(dirname $setup_py_file)
```

Reply:
Thanks for the tip! I don't know if I solved it the way you had envisioned but I got this to work:

```bash
SETUP_PY_FILE_PATTERN=$(ls -U **/$PKG_NAME/setup.py)
DIRECTORY_WITH_PACKAGE=$(dirname $SETUP_PY_FILE_PATTERN)
```

I found out that `ls -U` is faster because the results come unsorted, and for some reason doing `ls -U $SETUP_PY_FILE_PATTERN` didn't work but interpolating with other components _does_ work.

I guess there is a worry that `ls` could match multiple files? But hopefully not... in which case with only 1 file this works fine.

Reply:
The `ls -U` isn't doing anything, the glob gets evaluated in the shell before it even calls `ls`, so for example:

```sh
PKG_NAME=opentelemetry-instrumentation-flask
SETUP_PY_FILE_PATTERN=**/$PKG_NAME/setup.py
DIRECTORY_WITH_PACKAGE=$(dirname $SETUP_PY_FILE_PATTERN)
echo $DIRECTORY_WITH_PACKAGE
```

prints `instrumentation/opentelemetry-instrumentation-flask`

Reply:
> The ls -U isn't doing anything, the glob gets evaluated in the shell before it even calls ls

Ah thanks for that, I'll remove it right now

> prints `instrumentation/opentelemetry-instrumentation-flask`

Hm am I doing something wrong? I did the same but I get `**/opentelemetry-instrumentation-flask` printed out instead:

In my repo directory: `~/git/opentelemetry-python-contrib`

```
$ PKG_NAME=opentelemetry-instrumentation-flask
$ SETUP_PY_FILE_PATTERN=**/$PKG_NAME/setup.py
$ DIRECTORY_WITH_PACKAGE=$(dirname $SETUP_PY_FILE_PATTERN)
$ echo $DIRECTORY_WITH_PACKAGE
```

Prints `**/opentelemetry-instrumentation-flask`

Reply:
That's weird, what shell are you using? How about `*/$PKG_NAME/setup.py`?

Reply:
Ah i see what you mean. I'm not sure the best way, this seems to work OK:

```bash
$ find -wholename "*/$PKGNAME/setup.py"
./instrumentation/opentelemetry-instrumentation-flask/setup.py
$ find -wholename "*/$PKGNAME/foobar.py"
# prints nothing
```

I think `ls` would fail with the nonexistant file case because the glob doesn't expand:
```bash
$ ls */$PKGNAME/foobar.py
ls: cannot access '*/opentelemetry-instrumentation-flask/foobar.py': No such file or directory
```

I'm fine with whatever, just wanted to call out you could probably avoid the for loop 😃

Reply:
Yes thank you for the call out it reads and works much better without the loop! 😄 

Sounds good, I think the `ls` error could be useful for people debugging and it's not too noisy. It also runs faster than `find .` at least on my computer 🙂 

Either way on failure `SETUP_FILE_PATH` is empty so I've added this check for a better log message and early exit!

```
if [ -z $SETUP_PY_FILE_PATH ]; then
  echo "Error! setup.py not found for $PKG_NAME, can't build."
  exit -1
fi
```